core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -101,6 +101,8 @@ class LogCleanerIntegrationTest(compressionCodec: String) extends AbstractLogCle
       val messages = writeDups(numKeys = numKeys, numDups = 3, log = log, codec = codec)
       val startSize = log.size
 
+      log.onHighWatermarkIncremented(log.logEndOffset)
```

Comment:
Are these changes unrelated, or are these calls necessary to ensure that the high watermark is always available?